javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java

```diff
@@ -344,6 +344,9 @@ protected void buildUnexpectedResponseExceptionTypes(ProxyMethod.Builder builder
             swaggerExceptionDefinitions.exceptionTypeMapping);
         mergedExceptionTypeMapping.putAll(settingsExceptionTypeMap);
 
+        // remove expected status codes
+        expectedStatusCodes.forEach(mergedExceptionTypeMapping.keySet()::remove);
```

Comment:
@alzimmermsft Add this line to avoid generating (404 exists in both expected and unexpected, though azure-core seems handling it right)
```
        @Head("/http/success/404")
        @ExpectedResponses({204, 404})
        @UnexpectedResponseExceptionType(
                value = ClientAuthenticationException.class,
                code = {401})
        @UnexpectedResponseExceptionType(
                value = ResourceNotFoundException.class,
                code = {404})
        @UnexpectedResponseExceptionType(
                value = ResourceModifiedException.class,
                code = {409})
        @UnexpectedResponseExceptionType(HttpResponseException.class)
        Mono<Response<Boolean>> head404(
                @HostParam("$host") String host, RequestOptions requestOptions, Context context);
```